lib/rubocop/cop/lint/ambiguous_block_association.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for ambiguous block association with method
+      # when param passed without parentheses.
+      #
+      # @example
+      #
+      #   # bad
+      #
+      #   # It's ambiguous because there is no parentheses around `a` param
+      #   some_method a { |val| puts val }
+      #
+      # @example
+      #
+      #   # good
+      #
+      #   # With parentheses, there's no ambiguity.
+      #   some_method(a) { |val| puts val }
+      class AmbiguousBlockAssociation < Cop
+        MSG = 'Parenthesize the param `%s` to make sure that block will be '\
+              'associated with `%s` method call.'.freeze
+
+        def on_send(node)
+          _klass, method_name, args = node.children
+
+          return if method_name == :[]
+          return if node.parenthesized? || node.assignment?
+          return unless method_with_block?(args)
+
+          first_param = args.children.first
```

Comment:
@Drenmi thanks for review!
Unfortunately it doesn't work with `node.first_argument`. 
Here is example:
`node.source => "some_method a { |el| puts el }"`

Here is what we have in `node.first_argument`:
> => s(:block,
  s(:send, nil, :a),
  s(:args,
    s(:arg, :el)),
  s(:send, nil, :puts,
    s(:lvar, :el)))

If I use `args.children.first` I'll get:
> => s(:send, nil, :a)